ng/src/web/pages/agents/dialog.js

```diff
@@ -28,81 +29,95 @@ import {is_defined} from 'gmp/utils';
+
 const AgentDialog = ({
     agent,
-    comment,
-    name,
-    onValueChange
+    title,
+    visible,
+    onClose,
+    onSave,
   }) => {
 
   const is_edit = is_defined(agent);
 
   return (
-    <Layout flex="column">
-
-      <FormGroup title={_('Name')}>
-        <TextField
-          name="name"
-          grow="1"
-          value={name}
-          size="30"
-          onChange={onValueChange}
-          maxLength="80"/>
-      </FormGroup>
-
-      <FormGroup title={_('Comment')}>
-        <TextField
-          name="comment"
-          value={comment}
-          grow="1"
-          size="30"
-          maxLength="400"
-          onChange={onValueChange}/>
-      </FormGroup>
-
-      {!is_edit &&
-        <FormGroup title={_('Installer')}>
-          <FileField
-            name="installer"
-            onChange={onValueChange}/>
-        </FormGroup>
-      }
-
-      {!is_edit &&
-        <FormGroup title={_('Installer signature')}>
-          <FileField
-            name="installer_sig"
-            onChange={onValueChange}/>
-        </FormGroup>
-      }
-
-    </Layout>
+    <SaveDialog
+      visible={visible}
+      title={title}
+      onClose={onClose}
+      onSave={onSave}
+      initialData={agent}
+    >
+      {({
+        data: state,
+        onValueChange,
+      }) => {
+
+        return (
+          <Layout flex="column">
+
+            <FormGroup title={_('Name')}>
+              <TextField
+                name="name"
+                grow="1"
+                value={state.name}
+                size="30"
+                onChange={onValueChange}
+                maxLength="80"/>
+            </FormGroup>
+
+            <FormGroup title={_('Comment')}>
+              <TextField
+                name="comment"
+                value={state.comment}
+                grow="1"
+                size="30"
+                maxLength="400"
+                onChange={onValueChange}
+              />
+            </FormGroup>
+
+            {!is_edit &&
+              <FormGroup title={_('Installer')}>
+                <FileField
+                  name="installer"
+                  onChange={onValueChange}
+              />
+              </FormGroup>
+            }
+
+            {!is_edit &&
+              <FormGroup title={_('Installer signature')}>
+                <FileField
+                  name="installer_sig"
+                  onChange={onValueChange}
+                />
+              </FormGroup>
+            }
+          </Layout>
+        );
+      }}
+    </SaveDialog>
   );
 };
 
 AgentDialog.propTypes = {
   agent: PropTypes.model,
-  name: PropTypes.string,
-  comment: PropTypes.string,
+  title: PropTypes.string,
+  visible: PropTypes.bool,
+  onClose: PropTypes.func.isRequired,
+  onSave: PropTypes.func.isRequired,
   onValueChange: PropTypes.func,
 };
 
 
-export default withDialog({
-  title: _('New Alert'),
-  footer: _('Save'),
-  defaultState: {
-    comment: '',
-    name: _('Unnamed'),
```

Comment:
This is missing from the new implementation. You could set a default by using
```jsx
 <SaveDialog
     visible={visible}
     title={title}
     onClose={onClose}
     onSave={onSave}
     initialData={{name: _('Unnamed'), ...agent}}
>
```
Or even better set some variable like
```jsx
const defaults = {name: _('Unnamed')};
return (
 <SaveDialog
     visible={visible}
     title={title}
     onClose={onClose}
     onSave={onSave}
     initialData={{...defaults, ...agent}}
>
...
```
The second version is easier to extend later.